components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,230 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
 		checkTriggerRespMap = getBastionCheckResp(ss, bastionChecks, config)
 	}
 
-	// Get remote check trigger resp and append to checkTriggerRespMap (keys and values)
+	successfullyStartedMockServers, notStartedMockServers, err := ss.startMockServerIfNeeded(remoteChecks, config)
+	if err != nil {
+		ss.log.Error("Error occurred while starting mock server")
+		return models.BatchCheckResponse{}, err
+	}
+
+	if len(notStartedMockServers) > 0 {
+		for _, successfullyStartedMockServer := range successfullyStartedMockServers {
+			if successfullyStartedMockServer.Host != "" {
+				go ss.StopMockServerOnHostAndPort(successfullyStartedMockServer.Host, successfullyStartedMockServer.Protocol, successfullyStartedMockServer.Port)
+			}
+		}
+		return models.BatchCheckResponse{}, errors.New("mock server not started successfull on some nodes")
+	}
 	if len(remoteChecks) > 0 {
 		for key, value := range getRemoteCheckResp(ss, remoteChecks, config) {
 			checkTriggerRespMap[key] = value
 		}
 	}
-	return constructBatchCheckResponse(checkTriggerRespMap, append(bastionChecks, remoteChecks...))
+
+	if len(successfullyStartedMockServers) > 0 {
```

Comment:
Why not put this in a function and do it in `differ` when you start the mock servers, which will guarantee that it gets called at the end of this func BatchCheck ? 
Also then above logic to explicit stop will be avoided

Reply:
Cannot use defer as I am now collecting response from go routine and returning error is stop server fails. But I have moved code to a function for better readability.